client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java

```diff
@@ -376,14 +391,17 @@ private KeyPair getOrInitKeyPair() {
                 IOUtils.closeQuietly(privateKeyOutput);
                 IOUtils.closeQuietly(publicKeyOutput);
             }
+        } else {
+            LOG.debug("Error loading key pair!", "Key pair is not found and key strategy is default");
+            throw new KaaRuntimeException("Key pair is not found and your key strategy is default");
         }
         return keyPair;
     }
 
     @Override
     public EndpointKeyHash getEndpointKeyHash() {
         if (keyHash == null) {
-            EndpointObjectHash publicKeyHash = EndpointObjectHash.fromSHA1(getOrInitKeyPair().getPublic().getEncoded());
+            EndpointObjectHash publicKeyHash = EndpointObjectHash.fromSHA1(getOrInitKeyPair(false).getPublic().getEncoded());
```

Comment:
I insist on using <b>autogenerateKeys</b> variable here since we can't guarantee that _getOrIinitKeyPair_ won't be invoked here first time.